training-net/src/Repositories/Repository.cs

```diff
@@ -55,5 +55,7 @@ public void RemoveRange(IEnumerable<TEntity> entities)
         {
             Context.Set<TEntity>().RemoveRange(entities);
         }
+
+        public void Update(TEntity entity) => Context.Entry(entity).State = EntityState.Modified;

```

Comment:
you are updating the state of the entity but not its attributes ... is this working?